source/exe/hot_restart.cc

```diff
@@ -164,13 +164,16 @@ void HotRestartImpl::drainParentListeners() {
   shmem_.flags_ &= ~SharedMemory::Flags::INITIALIZING;
 }
 
-int HotRestartImpl::duplicateParentListenSocket(uint32_t port) {
+int HotRestartImpl::duplicateParentListenSocket(const std::string& address) {
   if (options_.restartEpoch() == 0) {
     return -1;
   }
 
   RpcGetListenSocketRequest rpc;
-  rpc.port_ = port;
+  if (address.length() >= sizeof(rpc.address_)) {
```

Comment:
not sure if I would bother with this check. strlcpy() will protect. I would just put in an ASSERT(address.length() < sizeof(rpc.address_))